app/scripts/lib/accounts/BalancesController.ts

```diff
@@ -154,27 +154,62 @@ export class BalancesController extends BaseController<
       },
     });
 
+    this.#tracker = new BalancesTracker(async (accountId: string) => {
+      // The BalancesTracker only uses account IDs, so we have to get the associated account first
+      const account = this.#listMultichainAccounts().find(
+        (multichainAccount) => multichainAccount.id === accountId,
+      );
+      if (!account) {
+        throw new Error(`Unknown account: ${accountId}`);
+      }
+      // Just to double-check, we make sure we are using a non-EVM account
+      if (!this.#isNonEvmAccount(account)) {
+        throw new Error(`Account is not a non-EVM account: ${accountId}`);
+      }
+
+      await this.#updateBalance(account);
```

Comment:
Maybe this could be a named function to avoid having all this logic in the constructor.